datastore/db/bytebase/FertiScan/update_inspection_function.sql

```diff
@@ -416,8 +440,8 @@ BEGIN
         FOR i IN 0..(max_length - 1)
         LOOP
             -- Extract values or set to empty string if not present
-            fr_value := COALESCE(fr_values->>i, '');
-            en_value := COALESCE(en_values->>i, '');
+            fr_value := fr_values->>i;
```

Comment:
How do we manage the edge case where both are Null? (also in the new_inspection)

Reply:
With a trigger. Please check file `datastore/db/bytebase/FertiScan/update_inspection_function.sql`

```sql
-- Function to check if both text_content_fr and text_content_en are NULL, and skip insertion if true
CREATE OR REPLACE FUNCTION "fertiscan_0.0.14".check_null_sub_label()
RETURNS TRIGGER AS $$
BEGIN
    -- Check if both text_content_fr and text_content_en are NULL
    IF NEW.text_content_fr IS NULL AND NEW.text_content_en IS NULL THEN
        -- Raise a notice and skip the insertion by returning NULL
        RAISE Exception 'Skipping insertion because both text_content_fr and text_content_en are NULL';
        RETURN NULL;
    END IF;
    -- Allow the insert if at least one is not NULL
    RETURN NEW;
END;
$$ LANGUAGE plpgsql;


-- Trigger to call check_null_sub_label() before inserting into sub_label
DROP TRIGGER IF EXISTS before_insert_sub_label ON "fertiscan_0.0.14".sub_label;
CREATE TRIGGER before_insert_sub_label
BEFORE INSERT ON "fertiscan_0.0.14".sub_label
FOR EACH ROW
EXECUTE FUNCTION check_null_sub_label();
```

In that case, we raise an exception as we agreed.

Reply:
Hmm, I would of checked for a double null value even before inserting to keep the operational cost as minimal as possible. Adding a trigger makes it more complex but create a safety net even outside the function, this solution works and I forgot we agreed upon it.

Thank you for pointing it out to me, This comment can be resolved.